pkg/apis/core/types.go

```diff
@@ -1966,6 +1966,28 @@ const (
 	PullIfNotPresent PullPolicy = "IfNotPresent"
 )
 
+// ContainerResizePolicy describes how container resources resize is handled.
+// Only one of the following container resize policies may be specified.
+// If none of the following policies is specified, it defaults to NoRestart.
+type ContainerResizePolicy string
+
+// These are the valid container resize policies
+const (
+	// Resize the container in-place without restarting it.
+	NoRestart ContainerResizePolicy = "NoRestart"
+	// Resize the container in-place by restarting it after resize.
+	RestartContainer ContainerResizePolicy = "RestartContainer"
+)
+
+// ResizePolicy represents the resource resize policy for a single container.
```

Comment:
Please be more precise about the meaning of this - is it a requirement that this container not be restarted or is it saying that a restart is not required (but may still happen if the kubelet needs to)?  Or something else.  I think this distinction may influence the constant strings above.

E.g. I *assume* this is a statement of capability.  "I can be resized in-place without restart".  That would lead me to look for something like:

```
resizePolicy:
  - resource: "cpu"
    restart: "NotRequired"
  - resource: "memory"
    restart: Required
```

What *other* policies do you think might one day be needed here?

Reply:
NoRestart tells Kubelet to call UpdateContainerResources CRI API to resize the resource - doing so does not restart the container. RestartContaienr policy on the other hand tell Kubelet to stop and start the container with new resources, and the use-case for this was some legacy java apps using -xmxN flag which are unable to use the increased memory without restarting.

We had RestartPod as a third option at one point, but removed it because we couldn't find a real-world use case for it other than the notion that someone might want to restart the entire pod in-place (to rerun the init containers).

I'll add more details to the comments.

Reply:
Are you thinking this should just be a binary value  - 'bool RestartOnResize' maybe? It certainly would make life a lot simpler.

Reply:
Hi, I found this half-completed, I don't know if it is still alive.

My point was to clarify whether "no restart" is contract or guidance.  If the kubelet, for some reason, needed to restart the pod, would that be a violation of NoRestart or just an unfortunate, but legal, edge case?

Reply:
I've updated [documentation](https://github.com/vinaykul/kubernetes/pull/1/commits/5126b9e1e93d2530ac820981c93986d64e46faa3#diff-4143fe2cdf42f1748648a9b80aa2ba07R1975) in the code to clarify that container resize policy is a guidance by stating that kubelet should resize the container without restarting it _if possible_ . I hope this is sufficient language.